kernel/src/hil/digest.rs

```diff
@@ -0,0 +1,48 @@
+//! Interface for Digest
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
```

Comment:
This should have additional callbacks for
- `clear_data`
- `add_data`

as these could be asynchronous depending on the HW. Maybe, a callback for setting the mode (and key) would be a good idea as well. If these are not asynchronous with OpenTitan, consider using a deferred call.

Reply:
Do we need a callback for clear data? What action are we expecting to do on the callback?

Reply:
Not necessarily an action, just an indication that the possibly long-running and asynchronous operation of clearing data or resetting the device has finished, and the client may proceed to calculate digests again.

Reply:
I haven't added a callback for `clear_data()` I'm not convinced we need one. Let me know what you think of the latest patches.

Reply:
I'm convinced one will be required sooner or later. Fine with me though, as soon as some HW comes around requiring it it must be added.